src/basic/chase-symlinks.c

```diff
@@ -57,22 +57,17 @@ static int log_autofs_mount_point(int fd, const char *path, ChaseSymlinksFlags f
                                  strna(n1), path);
 }
 
-int chase_symlinks(
-                const char *path,
-                const char *original_root,
-                ChaseSymlinksFlags flags,
-                char **ret_path,
-                int *ret_fd) {
-
-        _cleanup_free_ char *buffer = NULL, *done = NULL, *root = NULL;
-        _cleanup_close_ int fd = -1;
+int chase_symlinks_at(int dir_fd, const char *path, ChaseSymlinksFlags flags, char **ret_path, int *ret_fd) {
```

Comment:
i'd really break the arguments line here into multiple lines. Once we have 3 or so arguments it's just hard to read.

Reply:
what is "dir_fd"? is that the dir we take the path relative to? i.e. the *root* dir?